Add z-level aware version of Gui.drawRect

shadowfacts opened the pull request:
Adds a method which uses the z-level specified in the parameter and replaces the original method with a call to the new one using a z-level of 0, the previous hard-coded value.

LexManos commented:
Look at your patch.. What did you change, and why is there patches in there that shouldnt be?

shadowfacts commented:
I added a `zLevel` parameter to drawRect, used that instead of the previous hardcoded value of 0 in the `WorldRenderer#pos` calls, and added a new version of drawRect with the old parameters that called the updated method with 0 as the z-level.

Do you mean the parameter name changes? I presume those are because the parameters didn't get remapped to the old SRG values because the method signature changed.

LexManos commented:
Exactly, the method parameters cant change.
That breaks modders code.

shadowfacts commented:
I've replaced the method/param names (except `zLevel`) with those used in the original method.

LexManos commented:
It is standard procidure to add the old function on top because it makes smaller patch file.
Also, add the parameter to the end, not the middle of the arguments.

LexManos commented on patches/minecraft/net/minecraft/client/gui/Gui.java.patch:
```diff
@@ -0,0 +1,33 @@
+         func_73734_a(p_73728_1_, p_73728_2_ + 1, p_73728_1_ + 1, p_73728_3_, p_73728_4_);
+     }
+-
++    
```
?

LexManos commented on patches/minecraft/net/minecraft/client/gui/Gui.java.patch:
```diff
@@ -0,0 +1,33 @@
++    }
++
++    public static void func_73734_a(int p_73734_0_, int p_73734_1_, int p_73734_2_, int p_73734_3_, int p_73734_4_, double zLevel)
++    {
```
You're missing the exc/srg how are these being generated?

LexManos commented on patches/minecraft/net/minecraft/client/gui/Gui.java.patch:
```diff
@@ -0,0 +1,33 @@
++    
+     public static void func_73734_a(int p_73734_0_, int p_73734_1_, int p_73734_2_, int p_73734_3_, int p_73734_4_)
+     {
++    	func_73734_a(p_73734_0_, p_73734_1_, p_73734_2_, p_73734_3_, p_73734_4_, 0);
```
Spaces not tabs

shadowfacts commented:
What do you mean by "missing the exc"? I haven't touched the patch file, I've just made changes in Eclipse and run genPatches.

LexManos commented:
You should name the method and params in forge normally. 
And then gen the patches.
Then look at our custom srg and exc file and add the nessasary lines to it so that it generates THIS patch automatically.
The thing is, we push out the patches, and then FG remaps them.
If someone was to call this function and we DONT have the SRG file, they wont obfuscate properly.
And then there will be errors.
SOmeone should write up a tutorial on this.

LexManos commented on patches/minecraft/net/minecraft/client/gui/Gui.java.patch:
```diff
@@ -0,0 +1,29 @@
+ 
+     public static void func_73734_a(int p_73734_0_, int p_73734_1_, int p_73734_2_, int p_73734_3_, int p_73734_4_)
+     {
++    	drawRect(p_73734_0_, p_73734_1_, p_73734_2_, p_73734_3_, p_73734_4_, 0);
```
Spacing issue, use spaces not tabs.

LexManos commented on src/main/resources/forge.exc:
```diff
@@ -42,3 +42,5 @@ net/minecraft/server/management/ServerConfigurationManager.initializeConnectionT
 
 net/minecraft/stats/StatList.mergeStatBases([Lnet/minecraft/stats/StatBase;Lnet/minecraft/block/Block;Lnet/minecraft/block/Block;Z)V=|p_151180_0_,p_151180_1_,p_151180_2_,useItemIds
+
+net/minecraft/client/gui/Gui.drawRect(IIIIID)V=|p_73734_0_,p_73734_1_,p_73734_2_,p_73734_3_,p_73734_4_,zLevel
```
Put a new line here

shadowfacts commented:
I've fixed the whitespace issues and resolved the conflicts.

Actuarius commented:
@williewillus added labels [Feature]
